### ISSUE TITLE
Remove the silly BIGTABLE_CLIENT_NS macro.

### ISSUE THREAD

**Issue by coryan** [GoogleCloudPlatform/google-cloud-cpp#143]
Vote with a thumbs up if you would like the `BIGTABLE_CLIENT_NS` macro to go away in favor of the actual version (e.g., `v0` for now, `v7` some day).  Use a thumbs down if you want to keep it.

We would also add some tests to verify that the current version (as defined in `bigtable/client/version.h`) and namespaces match.

**Comment by coryan** [comment]
Any further opinions?

**Comment by mbrukman** [comment]
I think we should drop it completely, and not replace it with a `v0`, `v1`, etc. version.

Here's what other language clients do:

* [C#](https://github.com/GoogleCloudPlatform/google-cloud-dotnet/tree/master/apis): yes, but only to match the gRPC API, so right now, it uses V2 for Bigtable
* [Go](https://github.com/googlecloudplatform/google-cloud-go): yes, but only to match the gRPC API (Bigtable didn't follow the pattern, so there's no v2, but other projects use `apiv1` or `apiv2`, for example)
* [Java](https://github.com/GoogleCloudPlatform/google-cloud-java): no
* [Node.js](https://github.com/googleapis/nodejs-bigtable): no
* [PHP](https://github.com/GoogleCloudPlatform/google-cloud-php/tree/master/src/Bigtable/V2): yes, but only to match the gRPC API, so right now, it uses V2 for Bigtable
* [Python](https://github.com/GoogleCloudPlatform/google-cloud-python): no
* [Ruby](https://github.com/GoogleCloudPlatform/google-cloud-ruby/tree/master/google-cloud-bigtable/lib/google/bigtable/v2): yes, but only to match the gRPC API, so right now, it uses V2 for Bigtable

As a result, I think using a namespace which is based on a version number of the project, and not the gRPC version, will be completely different from what the other languages will do and cause confusion.

If we are concerned that folks will use submodules and include this code in their projects and then be broken due to incompatible versions or ODR violations, we should create tags, and they should sync to those, and upgrade carefully.

Was there a specific issue this namespace / versioning was intended to solve, that don't exist in other languages?

**Comment by coryan** [comment]
The objective of versioned namespaces is to allow an application to link against v2 of the library while another library they use links against v1 of the library (e.g., an application linking against Tensoflow whih might use v1 of the Cloud Bigable C++ client, and also against v2 of the Cloud Bigtable C++ client directly).  Without versioned namespaces that breaks in many ways (ODR to start with).

As to whether this problem exists in other languages:

* C# I am guessing yes, but pure guess.
* Go: they do not have this problem because they compile from source and link statically. 
* Java: do not know, but I would guess yes?
* Node.js, PHP, Python, Ruby solve this problem by using package managers to pick the right version for you and/or installing multiple versions side-by-side.

TL;DR; not a unique C++ problem, but more of a problem in C++.

**Comment by DanielMahu** [comment]
My concern is to what happens if said application links against a slightly patched, but still v1 version of our library. The advantage of having a macro name for the namespace is that it could be easily redefined when compiling to say e.g. MYAPP_v1.

**Comment by coryan** [comment]
Per the discussion with @sduskis today, we are going to keep this as-is.  It helps with diamond dependencies, and we can remove it at any time without much trouble to the users: source level compatibility is ~preserved, binary compatibility would be broken.
